meson_options.txt

```diff
@@ -1,6 +1,7 @@
 option('documentation', type: 'feature', description: 'Build documentation')
 option('html_manual', type: 'boolean', value: true, description: 'Build the HTML manual')
 option('manpages', type: 'boolean', value: true, description: 'Build manual pages')
+option('doxygen', type: 'boolean', value: false, description: 'Build doxygen source documentation')
```

Comment:
Maybe this should be a feature option instead, then doxygen can be looked up unconditionally and in the disabled case it will log "skipped: feature doxygen is disabled". Then you can handle the doc setup by simply checking if doxygen was found.

See how -Ddocmentation is handled for sphinx-build.

Reply:
Agree, that would be better, but since nobody really needs to build the docs (only those with an explicit preference), the default value should be "disabled" to reduce build overhead. The same could probably be done for the `documentation` flag. Nobody builds documentation (but me).